src/Microsoft.AspNetCore.TestHost/WebHostBuilderFactory.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.WebHostBuilderFactory;
+
+namespace Microsoft.AspNetCore.TestHost
+{
+    public static class WebHostBuilderFactory
+    {
+        public static IWebHostBuilder CreateFromStartup<TStartup>(string [] args)
```

Comment:
This is misleading. There's no direct link between TStartup and what gets run, it's only used to look up an assembly. Why not `CreateFromProgram<TProgram>`? That's what you're actually resolving and running. Then you could even bypass the entrypoint check and go directly to the method resolver.

Reply:
`CreateFromDefaultEntryPoint<TProgram>`

Reply:
I've been thinking about this for a bit. I think we should keep it this way (maybe find a better name like CreateFromStartupEntryPoint<TStartup> or something similar). The reason for this is that all our tools rely on the concept of Startup not the entry point (dotnet ef, etc.) and this method is meant to support the convention we follow in our templates (in which the startup class lives in the same assembly as the application).

Reply:
It's not right though. The entry point isn't necessarily where your startup code is. This is specifically about finding code in Program.cs, not the Startup class.

Reply:
Agreed, Startup hasn't been the primary entry point for a while now.